code/modules/mob/living/simple_animal/hostile/ooze.dm

```diff
@@ -0,0 +1,490 @@
+///Oozes are slime-esque creatures, they are highly gluttonous creatures primarily intended for player controll.
+/mob/living/simple_animal/hostile/ooze
+	name = "Ooze"
+	icon = 'icons/mob/vatgrowing.dmi'
+	icon_state = "gelatinous"
+	icon_dead = "gelatinous_dead"
+	mob_biotypes = MOB_ORGANIC
+	pass_flags = PASSTABLE | PASSGRILLE
+	ventcrawler = VENTCRAWLER_ALWAYS
+	gender = NEUTER
+	emote_see = list("jiggles", "bounces in place")
+	speak_emote = list("blorbles")
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	hud_type = /datum/hud/ooze
+	minbodytemp = 250
+	maxbodytemp = INFINITY
+	faction = list("slime")
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	health = 200
+	maxHealth = 200
+	attack_verb_continuous = "slimes"
+	attack_verb_simple = "slime"
+	attack_sound = 'sound/effects/blobattack.ogg'
+	a_intent = INTENT_HARM
+	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
+	mob_size = MOB_SIZE_LARGE
+	initial_language_holder = /datum/language_holder/slime
+
+	///Oozes have their own nutrition. Changes based on them eating
+	var/ooze_nutrition = 50
+	var/ooze_nutrition_loss = -0.15
+	var/ooze_metabolism_modifier = 2
+
+/mob/living/simple_animal/hostile/ooze/Initialize()
+	. = ..()
+	create_reagents(300)
+	add_cell_sample()
+	AddComponent(/datum/component/footstep, FOOTSTEP_MOB_SLIME, 7.5)
+
+/mob/living/simple_animal/hostile/ooze/attacked_by(obj/item/I, mob/living/user)
+	if(!check_edible(I))
+		return ..()
+	eat_atom(I)
+
+/mob/living/simple_animal/hostile/ooze/AttackingTarget(atom/A)
+	if(!check_edible(A))
+		return ..()
+	eat_atom(A)
+
+/mob/living/simple_animal/hostile/ooze/UnarmedAttack(atom/A)
+	if(!check_edible(A))
+		return ..()
+	eat_atom(A)
+
+///Handles nutrition gain/loss of mob and also makes it take damage if it's too low on nutrition, only happens for sentient mobs.
+/mob/living/simple_animal/hostile/ooze/Life()
+	. = ..()
+
+	if(!mind && stat != DEAD)//no mind no change
+		return
+
+	var/nutrition_change = ooze_nutrition_loss
+
+	//Eat a bit of all the reagents we have. Gaining nutrition for actual nutritional ones.
+	for(var/i in reagents.reagent_list)
+		var/datum/reagent/R = i
+		var/consumption_amount = min(reagents.get_reagent_amount(R.type), ooze_metabolism_modifier * REAGENTS_METABOLISM)
+		if(istype(R, /datum/reagent/consumable))
+			var/datum/reagent/consumable/consumable = R
+			nutrition_change += consumption_amount * consumable.nutriment_factor
+		reagents.remove_reagent(R.type, consumption_amount)
+	adjust_ooze_nutrition(nutrition_change)
+
+	if(ooze_nutrition <= 0)
+		adjustBruteLoss(0.5)
```

Comment:
Should all of this happen if the ooze is dead?

Reply:
good point and will do